This patch ensures that the 'tclIndex' files generated by 'auto_mkindex'
are sorted in a deterministic fashion.

Fixes a non-determinism issue reported
at <https://lists.gnu.org/archive/html/guix-devel/2015-10/msg00696.html>.

--- tcl8.6.4/library/auto.tcl	2015-02-26 17:57:28.000000000 +0100
+++ tcl8.6.4/library/auto.tcl	2015-11-13 23:18:34.964831717 +0100
@@ -207,6 +207,9 @@ proc auto_mkindex {dir args} {
 	set args *.tcl
     }
 
+    # Keep file names sorted in a determistic order.
+    set args [lsort -ascii $args]
+
     auto_mkindex_parser::init
     foreach file [glob -- {*}$args] {
 	try {
@@ -241,6 +244,10 @@ proc auto_mkindex_old {dir args} {
     if {![llength $args]} {
 	set args *.tcl
     }
+
+    # Keep file names sorted in a determistic order.
+    set args [lsort -ascii $args]
+
     foreach file [glob -- {*}$args] {
 	set f ""
 	set error [catch {
